### PR TITLE
[BrowserKit][Cache][Config][Console] Add missing return types

### PR DESCRIPTION
**Pull request by wouterj**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Ref https://github.com/symfony/symfony/issues/47551
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
(failures are related)

**Comment by wouterj** [comment]
Should be fixed now. The remaining failures are unrelated

**Review by stof — APPROVED** [review]

File: src/Symfony/Component/Config/Definition/PrototypedArrayNode.php [hunk 1]
```diff
@@ -156,6 +156,8 @@ public function getPrototype(): PrototypeNodeInterface
     /**
      * Disable adding concrete children for prototyped nodes.
      *
+     * @return never
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Config/Definition/PrototypedArrayNode.php:159]: I think we should use `@return void` here, and maybe review where we added those and revert them
the reason is that if the class is truely extensible, then we have no reason to force child classes to also throw here

Comment (wouterj) [reply src/Symfony/Component/Config/Definition/PrototypedArrayNode.php:159]: We can surely challenge whether this should not be an `@final` class, but I think also without it we should force new implementations to not have children. It does not make sense for a prototyped node to contain concrete children.

Comment (nicolas-grekas) [reply src/Symfony/Component/Config/Definition/PrototypedArrayNode.php:159]: ok, works for me then :)

**Comment by nicolas-grekas** [comment]
Thank you @wouterj.

File: src/Symfony/Component/Config/Definition/PrototypedArrayNode.php (hunk 1 above)
Comment (stof) [reply src/Symfony/Component/Config/Definition/PrototypedArrayNode.php:159]: This should indeed be `never` for that one (it overrides the parent method in a similar way we do in FrozenParameterBag overriding ParameterBag methods)

**Pull request merged**